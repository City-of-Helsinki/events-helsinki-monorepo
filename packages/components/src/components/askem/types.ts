import type { RnsData } from '../../types';
import type Askem from './Askem';

export type AskemBaseConfig = {
  disabled?: boolean;
  scriptUrl?: string;
  consentGiven?: boolean;
};

export type AskemConfigs = AskemBaseConfig & RnsData;

export interface AskemInstance {
  disabled: boolean;
  setRnsConfigValue: Askem['setRnsConfigValue'];
  consentGiven: boolean;
}
