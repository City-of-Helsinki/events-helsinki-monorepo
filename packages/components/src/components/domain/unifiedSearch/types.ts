import type {
  OpeningHours,
  OpeningHoursTimes,
  SearchListQueryVariables,
} from '../../../types';
import type { OrderByType, OrderDirType } from './unifiedSearchConstants';

export type UnifiedSearchParameters = {
  q?: string[];
  administrativeDivisionIds?: string[];
  ontologyTreeIds?: number[];
  ontologyWordIds?: number[];
  after?: string;
  first?: number;
  ontology?: string;
  isOpenNow?: boolean;
  openAt?: Date;
  orderBy?: OrderByType;
  orderDir?: OrderDirType;
};
export type UnifiedSearchVariables = SearchListQueryVariables;
export type UnifiedSearchOpeningHours = OpeningHours;
export type UnifiedSearchOpeningHoursTimes = OpeningHoursTimes;

export type TransitionOptions = {
  shallow?: boolean;
  locale?: string | false;
  scroll?: boolean;
};

export type ModifyFilters = (
  search: Partial<UnifiedSearchParameters>,
  transitionOptions?: TransitionOptions
) => void;
