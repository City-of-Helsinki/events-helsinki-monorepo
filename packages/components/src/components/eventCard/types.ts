import type { EventFields } from '../../types';

export type EventCardProps = {
  event: EventFields;
  eventUrl: string;
};

export type LargeEventCardProps = EventCardProps & {
  showEnrolmentStatusInCardDetails?: boolean;
};
