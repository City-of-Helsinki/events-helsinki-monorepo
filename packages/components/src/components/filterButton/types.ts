export type FilterType =
  | 'category'
  | 'hobbyType'
  | 'sportsCategory'
  | 'date'
  | 'dateType'
  | 'division'
  | 'place'
  | 'publisher'
  | 'target'
  | 'text'
  | 'minAge'
  | 'maxAge'
  | 'exactAge';
