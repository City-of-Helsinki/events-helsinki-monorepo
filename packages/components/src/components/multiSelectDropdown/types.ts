import type { Option } from '../../types/types';

export interface MultiselectDropdownProps {
  checkboxName: string;
  icon: React.ReactElement;
  inputPlaceholder?: string;
  inputValue?: string;
  name: string;
  onChange: (values: string[]) => void;
  options: Option[];
  fixedOptions?: Option[];
  renderOptionText?: (optionValue: string) => React.ReactChild;
  selectAllText?: string;
  setInputValue?: (newVal: string) => void;
  showSearch?: boolean;
  showSelectAll?: boolean;
  title: string;
  value: string[];
  className?: string;
  helpText?: string;
  filterByInput?: boolean;
}
