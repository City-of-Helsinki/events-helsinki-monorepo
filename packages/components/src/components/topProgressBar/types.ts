type LoggerFunction = (
  message?: string,
  ...optionalParameters: unknown[]
) => void;

export type Logger = {
  debug?: LoggerFunction;
  info?: LoggerFunction;
  warn?: LoggerFunction;
  error?: LoggerFunction;
};
