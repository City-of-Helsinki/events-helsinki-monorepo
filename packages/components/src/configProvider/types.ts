import type { ApolloClient, NormalizedCacheObject } from '@apollo/client';
import type { TFunction } from 'next-i18next';
import type { NextRouter } from 'next/router';

export interface Config {
  router?: NextRouter; // TODO: Support the react-router
  apolloClient?: ApolloClient<NormalizedCacheObject>;
  t: TFunction;
  /* eslint-disable @typescript-eslint/naming-convention */
  components: {
    A: (props: React.AnchorHTMLAttributes<HTMLAnchorElement>) => JSX.Element;
    Link?: (
      props: React.AnchorHTMLAttributes<HTMLAnchorElement>
    ) => JSX.Element;
  };
  /* eslint-enable @typescript-eslint/naming-convention */
}
