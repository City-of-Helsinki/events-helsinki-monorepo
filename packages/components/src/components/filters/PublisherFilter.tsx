import React from 'react';
import { useCommonTranslation, useErrorBoundary } from '../../hooks';
import { useOrganizationDetailsQuery } from '../../types';
import { FilterButton } from '../filterButton';
import type { FilterType } from '../filterButton';

interface Props {
  id: string;
  onRemove: (value: string, type: FilterType) => void;
}

const PublisherFilter: React.FC<Props> = ({ id, onRemove }) => {
  const { t } = useCommonTranslation();
  const { data, loading, error } = useOrganizationDetailsQuery({
    variables: { id },
  });
  useErrorBoundary(error);
  return loading ? (
    <FilterButton
      onRemove={onRemove}
      text={t('common:loading')}
      type="publisher"
      value={id}
    />
  ) : data && data.organizationDetails.name ? (
    <FilterButton
      onRemove={onRemove}
      text={data.organizationDetails.name}
      type="publisher"
      value={id}
    />
  ) : null;
};

export default PublisherFilter;
