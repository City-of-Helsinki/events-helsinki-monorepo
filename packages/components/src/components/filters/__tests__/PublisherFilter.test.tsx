import { translations } from '@events-helsinki/common-i18n';
import { render, waitFor, screen, userEvent } from '@/test-utils';
import { fakeOrganization } from '@/test-utils/mockDataUtils';
import { OrganizationDetailsDocument } from '../../../types';
import PublisherFilter from '../PublisherFilter';

const id = '1';
const name = 'Organization name';
const organization = fakeOrganization({ id, name });
const organizationResponse = { data: { organizationDetails: organization } };

const request = {
  query: OrganizationDetailsDocument,
  variables: {
    id,
  },
};

const mocks = [
  {
    request,
    result: organizationResponse,
  },
];

it('matches snapshot', async () => {
  const { container } = render(<PublisherFilter id={id} onRemove={vi.fn()} />, {
    mocks,
  });

  await screen.findByText(name);
  expect(container.firstChild).toMatchSnapshot();
});

it('calls onRemove callback when remove button is clicked', async () => {
  const onClickMock = vi.fn();
  render(<PublisherFilter id={id} onRemove={onClickMock} />, { mocks });

  await screen.findByText(name);

  await userEvent.click(
    screen.getByRole('button', {
      name: translations.common.filter.ariaButtonRemove.replace(
        '{{filter}}',
        name
      ),
    })
  );

  expect(onClickMock).toHaveBeenCalled();
  expect(onClickMock).toHaveBeenCalledWith(id, 'publisher');
});

it("should return null if place doesn't exist", async () => {
  const mocks = [
    {
      request,
      error: new Error('not found'),
    },
  ];

  const { container } = render(<PublisherFilter id={id} onRemove={vi.fn()} />, {
    mocks,
  });

  await waitFor(() => {
    expect(
      screen.queryByText(translations.common.loading)
    ).not.toBeInTheDocument();
  });

  expect(container.innerHTML).toBe('');
});
