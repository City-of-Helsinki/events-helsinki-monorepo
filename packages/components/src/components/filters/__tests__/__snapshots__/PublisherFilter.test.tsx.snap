// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="Tag-module_tag__3bo5S tag_hds-tag__3MKf3 custom-theme-1 Tag-module_tag-rounded-corners__mCq3C tag_hds-tag--rounded-corners__2KR20 filter"
  id="hds-tag"
>
  <span
    class="Tag-module_label__zCgFr tag_hds-tag__label__I6mc8"
    id="hds-tag-label"
  >
    <span
      aria-hidden="true"
    >
      Organization name
    </span>
  </span>
  <button
    aria-label="Poista suodatin: Organization name"
    class="Tag-module_deleteButton__1diMR tag_hds-tag__delete-button__33Tgz tag_hds-tag__delete-button__33Tgz"
    id="hds-tag-delete-button"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <rect
          height="24"
          width="24"
        />
        <polygon
          fill="currentColor"
          points="18 7.5 13.5 12 18 16.5 16.5 18 12 13.5 7.5 18 6 16.5 10.5 12 6 7.5 7.5 6 12 10.5 16.5 6"
        />
      </g>
    </svg>
  </button>
</div>
`;
